src/features/ProfilePage/components/EmailEditor.tsx

```diff
@@ -11,28 +12,31 @@ import { IconButton, TextButton } from '@/components/Buttons';
 import LabeledInput from '@/components/LabeledInput';
 import Text from '@/components/Text';
 
+import type { Account } from '@/features/Authentication/authenticationApi';
+
 const EmailEditor = () => {
   const { t } = useTranslation('profile');
-  const userInfo = useAppSelector(selectUserInfo);
+  const account = useAppSelector(selectAccount);
+  const [updateAccount, { isLoading, isSuccess }] = useUpdateAccountMutation();
 
+  const [email, setEmail] = useState(account.email);
   const [isOpen, setIsOpen] = useState(false);
   const toggleIsOpen = () => setIsOpen(!isOpen);
-  const [email, setEmail] = useState<string>('');
 
   const isEmailMissing = !email.length;
   const emailValue = isEmailMissing ? t('account.email.missing') : email;
   const isEmailInvalid = email.length > 0 && !email.match(EMAIL_REGEX);
-  const isSavingDisabled = isEmailMissing || isEmailInvalid;
-
-  useEffect(() => {
-    setEmail(userInfo.email ?? '');
-  }, [userInfo.email]);
+  const isSavingDisabled = isLoading || isEmailMissing || isEmailInvalid;
 
-  const saveNewEmail = () => {
-    console.log('API: Save new email');
-    setIsOpen(false);
+  const accountToSave: Account = {
+    ...account,
+    email,
   };
 
+  useEffect(() => {
+    if (isSuccess) setIsOpen(false);
+  }, [isSuccess]);
+
```

Comment:
I think I might prefer the .unwrap pattern instead of these useEffects.
IMO its more readable.
useEffect causes extra brainwork (at least for me)..

so..

```typescript
const handleAction = async () => {
  try{
    await updateAccount().unwrap()
    closeModal()
  } catch(error)
     // we dont need to do anything here? since we already launch the notifications from the api-client 
     // onQueryStarted thingymajingy.. but maybe good to test. 
}
        ```

Reply:
I don't really see a huge difference here, but I changed it. I guess it's nice to have fewer component dependencies (no useEffect and isSuccess), but I really can't judge, which is better. 🤔 🤷‍♂️